Source/Core/Resource.js

```diff
@@ -126,6 +130,34 @@ define([
         request.deferred = undefined;
     }
 
+    /**
+     * @private
+     */
+    function combineQueryParameters(q1, q2, preserveQueryParameters) {
```

Comment:
Please add some comments about what this function is actually doing (i.e. perhaps just some examples of expected input/output behavior).  The intent of these types of functions are a pain to figure out once you haven't looked at it in a while (or are seeing it for the first time).